proposals/NNNN-lifetime-dependency.md

```diff
@@ -0,0 +1,481 @@
+  In this case, for reasons explained earlier, we default to copying the lifetime dependency.
+  If a scoped lifetime dependency is needed, it can be explicitly requested by adding the `scoped` keyword:
+  
+```swift
+func f(arg: borrowing ArgType) -> @dependsOn(scoped arg) ResultType
+```
+
+**Methods:** Similar rules apply to `self` lifetime dependencies on methods.
+Given a method of this form:
+
+```swift
+<mutation-modifier> func method(... args ...) -> @dependsOn(self) ResultType
+```
+
+The behavior depends as above on the mutation-modifier and whether the defining type is `Escapable` or `~Escapable`.
+
+**Initializers:** An initializer can define a lifetime dependency on one of its arguments.
+In this case, we use the same rules same as for “Functions” above
+by using the convention that initializers can be viewed as functions that return `Self`:
+
+```swift
+init(arg: <parameter-convention> ArgType) -> @dependsOn(arg) Self
+```
+
+### Implicit Lifetime Dependencies
+
+The syntax above allows developers to explicitly annotate lifetime dependencies in their code.
+But because the possibilities are limited, we can usually allow the compiler to infer a suitable dependency.
+The detailed rules are below, but generally we require that the return type be `~Escapable` and that there be one “obvious” source for the dependency.
+
+In particular, we can infer a lifetime dependency on `self` for any method that returns a `~Escapable` type.
+As above, the details vary depending on whether `self` is `Escapable` or `~Escapable`:
+
+```swift
+struct NonescapableType: ~Escapable { ... }
+struct EscStruct {
+  func f1(...) -> /* @dependsOn(self) */ NonescapableType
+  borrowing func f2(...) -> /* @dependsOn(self) */ NonescapableType
+  mutating func f3(...) -> /* @dependsOn(self) */ NonescapableType
+
+  // Note: no lifetime dependency is inferred for a consuming method
+  // on an `Escapable` type, since there is no legal option
+  consuming func f4(...) -> NonescapableType
+}
+
+struct NEStruct: ~Escapable {
+  func f1(...) -> /* @dependsOn(self) */ NonescapableType
+  borrowing func f2(...) -> /* @dependsOn(self) */ NonescapableType
+  mutating func f3(...) -> /* @dependsOn(self) */ NonescapableType
+
+  // Note: A copy lifetime dependency is legal here
+  consuming func f4(...) -> /* @dependsOn(self) */ NonescapableType
+}
+```
+
+For free or static functions or initializers, we can infer a lifetime dependency when the return value is `~Escapable` and there is only one obvious argument that can serve as the source of the dependency.
+For example:
+
+```swift
+struct NEType: ~Escapable { ... }
+
+// If there is only one argument with an explicit parameter convention:
+func f(..., arg1: borrowing Type1, ...) -> /* @dependsOn(arg1) */ NEType
+
+// Or there is only one argument that is `~Escapable`:
+func g(..., arg2: NEType, ...) -> /* @dependsOn(arg2) */ NEType
```

Comment:
We don't infer when mutation modifier is absent. arg2 doesn't have one.

Reply:
@atrick and I discussed permitting this as long as:
* the argument was nonescapable, AND
* the return type was nonescapable

Reply:
This is how it's implemented right now. 

I thought we decided not to rely on default ownership atm. @atrick